traitsui/tests/editors/test_range_editor.py

```diff
@@ -49,3 +49,70 @@ def test_simple_editor_format_func(self):
 
     def test_custom_editor_format_func(self):
         self.check_range_enum_editor_format_func("custom")
+
+    @requires_toolkit([ToolkitName.qt])
+    def check_set_with_text(self, mode):
+        model = RangeModel()
+        view = View(
+            Item(
+                "value",
+                editor=RangeEditor(low=1, high=12, mode=mode)
+            )
+        )
+        tester = UITester()
+        with tester.create_ui(model, dict(view=view)) as ui:
+            number_field = tester.find_by_name(ui, "value")
+            text = number_field.locate(locator.WidgetType.textbox)
+            text.perform(command.KeySequence("\b\b\b\b\b4"))
+            text.perform(command.KeyClick("Enter"))
+            displayed = text.inspect(query.DisplayedText())
+            self.assertEqual(model.value, 4)
+            self.assertEqual(displayed, str(model.value))
+
+    def test_simple_slider_editor_set_with_text(self):
+        return self.check_set_with_text(mode='slider')
+
+    def test_large_range_slider_editor_set_with_text(self):
+        return self.check_set_with_text(mode='xslider')
+
+    def test_log_range_slider_editor_set_with_text(self):
+        return self.check_set_with_text(mode='logslider')
+
+    def test_range_text_editor_set_with_text(self):
+        return self.check_set_with_text(mode='text')
+
+    # There is a problem with KeySequence on wx.  trying to include the key
+    # '\b' does not succesfully type a backspace.  Instead EmulateKeyPress
+    # seems to literally type "\x08" which leads to errors.
```

Comment:
That makes sense so me, but would we then need to hard code all other escape characters?  for example '\b', '\t', '\n', etc.  That may be just what we need to do and that is perfectly fine.  I had just left the tests separate for now. so that the issue was known and we could try to address it.  It seems very likely to me that this may be the only solution, but I didn't know if there may would be a way to get the escape characters to work directly without explicitly checking and swapping them for the pyface key names.

Reply:
There are only a few of these characters so I think it is not a big deal: We can maintain this as a separate mapping. Alternatively we can forbid the use of `
\b` in KeySequence and force test code to use KeyClick("Backspace"), but we should choose one route, not both.

Reply:
`KeyClick("Backspace")` with an optional argument for the # of backspaces sounds a lot better than manually doing `\b`.

Reply:
OK, then we would have to check for `\b` in the Qt implementation to disallow it. Otherwise test code trying to use `\b` would not be toolkit agnostic. In either case, these escape characters need to be handled one way or another.

Reply:
Just to clarify, I think either way is fine.

Reply:
> `KeyClick("Backspace")` with an optional argument for the # of backspaces sounds a lot better than manually doing `\b`.

It makes sense to me to support both.  My understanding is just that we want the code to be toolkit agnostic.  I added kits suggestion for now so that '\b' works on both wx and qt.  Maybe in a later pr we can add to the KeyClick functions to take an optional argument to repeat the specified character n times.